jobs/contact-retrieve-transcript/exportTranscript.ts

```diff
@@ -35,11 +37,91 @@ export const exportTranscript = async ({
     type: m.type,
     media: m.media,
   }));
+};
+
+const getUser = async (client: ReturnType<typeof getClient>, serviceSid: string, from: string) => {
+  try {
+    return await client.chat.v2
+      .services(serviceSid)
+      .users.get(from)
+      .fetch();
+  } catch (err) {
+    if (err instanceof RestException && err.code === 20404) {
+      return null;
+    }
+    throw err;
+  }
+};
+
+const getRole = async (
+  client: ReturnType<typeof getClient>,
+  serviceSid: string,
+  roleSid: string,
+) => {
+  try {
+    return await client.chat.v2
+      .services(serviceSid)
+      .roles.get(roleSid)
+      .fetch();
+  } catch (err) {
+    if (err instanceof RestException && err.code === 20404) {
+      return null;
+    }
+    throw err;
+  }
+};
+
+const getParticipants = async (
+  client: ReturnType<typeof getClient>,
+  serviceSid: string,
+  messages: Awaited<ReturnType<typeof getTransformedMessages>>,
+) => {
+  const froms: Array<string> = [];
+
+  const participants: ExportTranscripParticipants = {};
+
+  messages.forEach(m => {
+    if (!froms.includes(m.from)) {
+      froms.push(m.from);
+    }
+  });
+
+  const promises = froms.map(async from => {
+    const user = await getUser(client, serviceSid, from);
+    const role = user?.roleSid ? await getRole(client, serviceSid, user.roleSid) : null;
+
+    participants[from] = {
+      user,
```

Comment:
@GPaoloni Your spec asked for a couple of specific fields, but I didn't see any harm in returning *everything* that twilio gives us in case we need something else later.

Also you asked for `isCounsellor`, but neither of these endpoints returned anything like that. You should be able to write a comparison on the front end to compare `role.friendlyName` to figure out if they are a counselor or not.

Reply:
>Your spec asked for a couple of specific fields, but I didn't see any harm in returning everything that twilio gives us in case we need something else later.

I'm not firmly against this, since I see no harm, but also there's no benefit storing stuff like `isOnline` or the creation/update dates probably. Again since there's no harm anyways I am not asking for a change!

>Also you asked for isCounsellor, but neither of these endpoints returned anything like that. You should be able to write a comparison on the front end to compare role.friendlyName to figure out

Yes I thought this was very likely to happen, but shouldn't be hard to calculate that in the front end.
The only case I am wondering tho is if a role is ever removed/renamed, we won't be able to do the calculation at read time. Maybe that should be addressed here (setting `isCounellor` on export time?). This is more of a question.